salt/states/pip_state.py

```diff
@@ -852,7 +852,7 @@ def installed(
             )
         # If we fail, then just send False, and we'll try again in the next function call
         except Exception as exc:  # pylint: disable=broad-except
-            log.exception(exc)
+            globals().get("log").exception(exc)
```

Comment:
can this be `logging.getLogger().exception(exc)` Also, logging `exc` seems redundant since the exception log handler already includes info about the exception. Adding a useful string about what is happening when the exception is being handled is better IMO.

Reply:
I would prefer renaming the global `log` object to, for example, `logger`, rather that this `globals()` approach.

The existing log call is `log.exception(exc)` which, as @dwoz pointed out, is not the best approach.

Something like, `log.exception("Some useful message: %s", exc)`

This would log the first line of the exception along with the message, and then the whole traceback.